components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/provisioning/impl/DefaultProvisioningHandler.java

```diff
@@ -718,7 +719,42 @@ private String getUserStoreDomain(String userStoreDomain, UserRealm realm)
      * @return
      */
     protected String generatePassword() {
-        return RandomStringUtils.randomNumeric(12);
+
+        // Pick from some letters that won't be easily mistaken for each other.
+        // So, for example, omit o O and 0, 1 l and L.
+        // This will generate a random password which satisfy the following regex.
+        // ^((?=.*\\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%&*])).{12}$}
+        Random secureRandom = new SecureRandom();
+        String characters = "23456789abcdefghjkmnpqrstuvwxyzABCDEFGHJKMNPQRSTUVWXYZ!@#$%&*";
+        String digits = "23456789";
+        String lowercaseLetters = "abcdefghjkmnpqrstuvwxyz";
+        String uppercaseLetters = "ABCDEFGHJKMNPQRSTUVWXYZ";
+        String specialCharacters = "!@#$%&*";
```

Comment:
Instead of writing the entire character set for characters, how about the following approach :
`String characters = digits + lowercaseLetters + uppercaseLetters + specialCharacters;`

Reply:
Addressed in the latest commit https://github.com/wso2/carbon-identity-framework/pull/5590/commits/e058432bae954566873729e3974a87a814c4d675